doc/source/data/data-internals.rst

```diff
@@ -1,42 +1,79 @@
+Ray Data uses either :ref:`Ray tasks <task-key-concept>` or
+:ref:`Ray actors <actor-key-concept>` to transform blocks. By default, it uses tasks.
 
-To ensure CPU resources are always available for Ray Data execution, limit the number of concurrent Tune trials. This can be done using the ``max_concurrent_trials`` Tune option.
+.. image:: images/dataset-map.svg
+   :align: center
+..
+  https://docs.google.com/drawings/d/12STHGV0meGWfdWyBlJMUgw7a-JcFPu9BwSOn5BjRw9k/edit
 
-.. literalinclude:: ./doc_code/key_concepts.py
-  :language: python
-  :start-after: __resource_allocation_1_begin__
-  :end-before: __resource_allocation_1_end__
+For more information on transforming data, see
+:ref:`Transforming data <transforming_data>`.
 
-.. _datasets_pg:
+Shuffling data
+--------------
 
-Ray Data and Placement Groups
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+When you call :meth:`~ray.data.Dataset.random_shuffle`,
+:meth:`~ray.data.Dataset.sort`, or :meth:`~ray.data.Dataset.groupby`, Ray Data shuffles
+blocks in a map-reduce style: map tasks partition blocks by value and then reduce tasks
+merge co-partitioned blocks.
 
-By default, Ray Data configures its tasks and actors to use the cluster-default scheduling strategy ("DEFAULT"). You can inspect this configuration variable here:
-:class:`ray.data.DataContext.get_current().scheduling_strategy <ray.data.DataContext>`. This scheduling strategy will schedule these tasks and actors outside any present
-placement group. If you want to force Ray Data to schedule tasks within the current placement group (i.e., to use current placement group resources specifically for Ray Data), you can set ``ray.data.DataContext.get_current().scheduling_strategy = None``.
+.. note::
 
-This should be considered for advanced use cases to improve performance predictability only. We generally recommend letting Ray Data run outside placement groups as documented in the :ref:`Ray Data and Other Libraries <datasets_tune>` section.
+    Shuffles materialize :class:`Datasets <ray.data.Dataset>` in memory. In other
+    words, shuffle execution isn't streamed through memory.
+
+For an in-depth guide on shuffle performance, see :ref:`Performance Tips and Tuning <shuffle_performance_tips>`.
+
+Scheduling
+==========
+
+Ray Data uses Ray core for execution, and hence is subject to the same scheduling considerations as normal Ray tasks and actors. Ray Data uses the following custom scheduling settings by default for improved performance:
+
+* The ``SPREAD`` scheduling strategy is used to ensure data blocks are evenly balanced across the cluster.
+* Retries of application-level exceptions are enabled to handle transient errors from remote datasources.
+* Dataset tasks ignore placement groups by default, see :ref:`Ray Data and Placement Groups <datasets_pg>`.
```

Comment:
Did we still want to keep the additional details about Ray Data interaction with placement groups and Tune?